test/lib/utils/addLiquidityHelper.ts

```diff
@@ -340,21 +337,30 @@ export function assertAddLiquidityProportional(
     const { txOutput, addLiquidityQueryOutput, addLiquidityBuildCallOutput } =
         addLiquidityOutput;
 
-    const bptToken = new Token(
-        addLiquidityInput.chainId,
-        poolState.address,
-        18,
+    // replace referenceAmount into amountsIn or bptOut for comparison
+    const bptOut = addLiquidityQueryOutput.bptOut.token.isSameAddress(
+        addLiquidityInput.referenceAmount.address,
+    )
+        ? TokenAmount.fromRawAmount(
+              addLiquidityQueryOutput.bptOut.token,
+              addLiquidityInput.referenceAmount.rawAmount,
+          )
+        : addLiquidityQueryOutput.bptOut;
+    const amountsIn = addLiquidityOutput.addLiquidityQueryOutput.amountsIn.map(
+        (a) =>
+            a.token.isSameAddress(addLiquidityInput.referenceAmount.address) &&
+            !a.token.isSameAddress(poolState.address) // skip CSP BPT as token
+                ? TokenAmount.fromRawAmount(
+                      a.token,
+                      addLiquidityInput.referenceAmount.rawAmount,
+                  )
+                : a,
     );
 
-    const expectedQueryOutput: Omit<
-        AddLiquidityQueryOutput,
-        'amountsIn' | 'bptIndex'
-    > = {
-        // Query should use same bpt out as user sets
-        bptOut: TokenAmount.fromRawAmount(
-            bptToken,
-            addLiquidityInput.bptOut.rawAmount,
-        ),
+    const expectedQueryOutput: Omit<AddLiquidityQueryOutput, 'bptIndex'> = {
+        // Query should use same referenceAmount as user sets
+        bptOut,
```

Comment:
Similar to above - seems like we're not really testing bptOut here?

Reply:
We are comparing that to referenceAmount whenever reference amount is the bptOut. On other cases, I fill out with query amounts so it passes. Again, seems easier then adapting getCheck with/without bptOut depending on the input content.
If you think it's too confusing as is, I can spend some time on an alternative.